lib/ansiblelint/__main__.py

```diff
@@ -125,8 +126,6 @@ def main() -> int:
     """Linter CLI entry point."""
     cwd = pathlib.Path.cwd()
 
-    options = cli.get_config(sys.argv[1:])
```

Comment:
Removing this doesn't look right. Look for something that may be mutating this list (it's not immutable).

Reply:
@webknjaz we are not removing it, it is happening inside the `config` module. The only issue that we have is that  now we need options to be loaded inside `utils` module, and this affected test suites where we used to test different by just importing the methods we wanted to test. This now triggers config loading as side effect and fails because the command line arguments are not the same.

We could likely avoid it by mocking the import or mocking the cli arguments.

A longer term solution would be to move stuff outside the utils module which used to be the kitchensink.

Reply:
That doesn't look right either. The typical idiom is to access `sys.argv` on the top level (that would be this module). The main function is supposed to accept it as an argument and the outer func would access `sys.argv` and do return `main(sys.argv)`. That outer func would also have to be specified as a console script entrypoint.

This way, anybody can cleanly call `main()` with any args they want from tests w/o having to worry about the state of the global `sys.argv`.

Reply:
Please check again, I rewrote it.